flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBStateDownloader.java

```diff
@@ -90,27 +80,62 @@ private void downloadDataForAllStateHandles(
             } else {
                 throw new FlinkRuntimeException("Failed to download data for state handles.", e);
             }
+        } finally {
+            // Unregister and close the internal closer. In a failure case, this should interrupt
+            // ongoing downloads.
+            closeableRegistry.unregisterCloseable(internalCloser);
+            IOUtils.closeQuietly(internalCloser);
```

Comment:
I think it should be 
```
if (closeableRegistry.unregisterCloseable(internalCloser)) {
    IOUtils.closeQuietly(internalCloser);
}
```
to prevent double close.